packages/shared/src/keys.ts

```diff
@@ -110,10 +110,14 @@ export function isProductionFromSecretKey(apiKey: string): boolean {
   return apiKey.startsWith('live_') || apiKey.startsWith('sk_live_');
 }
 
-export const getCookieSuffix = (publishableKey: string): string => {
-  return publishableKey.split('_').pop() || '';
-};
+export async function getCookieSuffix(publishableKey: string): Promise<string> {
+  const data = new TextEncoder().encode(publishableKey);
+  const digest = await globalThis.crypto.subtle.digest('sha-1', data);
+  const stringDigest = String.fromCharCode(...new Uint8Array(digest));
+  // Base 64 Encoding with URL and Filename Safe Alphabet: https://datatracker.ietf.org/doc/html/rfc4648#section-5
+  return isomorphicBtoa(stringDigest).replace(/\+/gi, '-').replace(/\//gi, '_').substring(0, 8);
```

Comment:
This could result in cookie names that look like `__session_a3f_bd3f`
This is not going to play well with `split('_')` from the base branch because:

```
"__session_asdfasdf".split('_').pop();
> 'asdfasdf'
"__session_asdfa_sdf".split('_').pop();
> 'sdf'
```

In your research, did you hit any case where we can end up with cookie suffixes less than 8 characters long?
If not, I suggest we stick with the initial proposal of simply stripping `+`, `/` and `=`. Or we can replace them with an alphanumeric char like `0`. This is going to lower entropy but just marginally so we shouldn't worry about it

Reply:
As discussed online, since we don't use the `.split('_')` and if choose to use a delimiter we would prefer a symbol outside of the currently used in cookie names (eg `:` because `_` is used as word separator in cookie names) we will keep the url-safe base64 solution.
cc: @nikosdouvlis

Reply:
Agreed, marking this as resolved :)